docs/DesignDocs/SGX_EMM.md

```diff
@@ -0,0 +1,350 @@
+SGX Enclave Memory Manager
+=================================
+
+## Motivation ##
+
+An enclave's memory is backed by special reserved region in RAM, called Enclave Page Cache(EPC). Enclave memory management tasks include allocating/reserving virtual address ranges, committing physical EPC pages, changing EPC page permissions or page type, or removing EPC pages. Those tasks require collaboration between trusted runtime, untrusted runtime and the OS. The SGX enclave memory manager (EMM) serves as a central component in the enclave trusted runtime that abstracts the interaction with untrusted runtime for all memory management flows and provides APIs for its clients to reserve virtual address ranges, commit EPC memory to the reserved address ranges, modify attributes of the reserved/committed pages.   
```

Comment:
nit: please split lines into (say) 80 column width, which makes git diffs easier to follow.

Reply:
will do for next version